src/UkModulusCheck/Implementation.fs

```diff
@@ -0,0 +1,232 @@
+namespace UkModulusCheck
+
+module internal Implementation =
+    open System
+    open Helpers
+    open Types
+
+    module Position =
+        let U, V, W, X, Y, Z, A, B, C, D, E,  F,  G,  H =
+            0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13
+
+    let parseMethod = function
+        | "MOD10" -> Some Standard10
+        | "MOD11" -> Some Standard11
+        | "DBLAL" -> Some DoubleAlternate
+        | _ -> None
+
+    let parseValidationRule (line: string) =
+        match line.Split([|' '|], StringSplitOptions.RemoveEmptyEntries) with
+        | xs when xs.Length = 18 ->
+            let scFrom, scTo = xs.[0], xs.[1]
+            let method = xs.[2] |> parseMethod
+            let weightings = xs.[3..16] |> Array.toList |> List.map str2int |> List.sequence
+            let ex = str2int xs.[17]
+            Option.map3 (fun m ws e ->
+                { SortCodeFrom = scFrom; SortCodeTo = scTo; Method = m; Weightings = ws; Exception = Some (Exception e) })
+                method weightings ex
+        | xs when xs.Length = 17 ->
+            let scFrom, scTo = xs.[0], xs.[1]
+            let method = xs.[2] |> parseMethod
+            let weightings = xs.[3..16] |> Array.toList |> List.map str2int |> List.sequence
+            Option.map2 (fun m ws ->
+                { SortCodeFrom = scFrom; SortCodeTo = scTo; Method = m; Weightings = ws; Exception = None })
+                method weightings
+        | _ -> None
+
+    let parseSubstitution (line: string) =
+        match line.Split([|' '|], StringSplitOptions.RemoveEmptyEntries) with
+        | [|scFrom; scTo|] -> Some { SortCode = scFrom; SubstituteWith = scTo }
+        | _ -> None
+
+    let standardise (sortCode: string) (accountNo: string) =
+        match sortCode.Length, accountNo.Length with
+        | 6, 8 -> Some (sortCode, accountNo)
+        | 6, 6 -> Some (sortCode, "00" + accountNo)
+        | 6, 7 -> Some (sortCode, "0" + accountNo)
+        | 6, 9 -> Some (sortCode.[0..4] + accountNo.[0..0], accountNo.[1..]) // For Santander
+        | 6, 10 ->
+            if sortCode.StartsWith("08") // Co-Operative Bank plc
+            then Some (sortCode, accountNo.[..7])
+            else Some (sortCode, accountNo.[2..])
+        | _ -> None
+
+    let doubleAlternateModulus (weightings: Weight list) (number: string) : int =
+        number
+        |> Seq.map char2int
+        |> Seq.map2 (*) weightings
+        |> Seq.collect (Seq.unfold (fun x -> if x > 0 then Some (x % 10, x / 10) else None))
```

Comment:
I'd think about making this bit simpler, even if it would take more code.

Reply:
Any ideas how to simplify it? The code is pretty straightforward for me and hardly can see how to make it simpler, but I may biased having implemented it ;)